### DISCUSSION TITLE
Online tool for debugging and issue replication

### DISCUSSION THREAD

**Discussion by jackbentley** [PHP-CS-Fixer/PHP-CS-Fixer#8232] [Feature ideas]
Would it be possible to create an online try/demo/playground tool which takes a config and input and shows the output running on latest version?

Similar to https://phpstan.org/try or https://getrector.com/demo

I've been interacting with these projects and using these tools report bugs/test/etc. It makes the whole issue flow a lot easier. It makes it really easy and simple for every issue to have a reproducer which can be tested on latest version easily way after the issue is created - meaning backlog management is a lot easier. It also puts less pressure on the issue author to come back and check/update/close/etc the issue afterwards (which we know doesn't always happen).

Having to continuously manually update and check old issues I've created here is a bit of a pain. The tools currently in place will automatically close issues when no one had responded presumably because there's no simple way to check if something is still an issue or not. With an online tool like suggested you could even update that bot to check if output has changed since the issue was created and notify of intention to close or something?

Both mentioned tools have their implementation open-sourced:

PHPStan
 - https://github.com/phpstan/phpstan/tree/2.0.x/playground-api
 - https://github.com/phpstan/phpstan/tree/2.0.x/playground-runner
 - https://github.com/phpstan/phpstan/blob/2.0.x/website/src/try.njk
 - https://github.com/phpstan/phpstan/blob/2.0.x/website/src/js/PlaygroundViewModel.ts

Rector
 - https://github.com/rectorphp/getrector-com/blob/main/src/DemoRunner.php
 - https://github.com/rectorphp/getrector-com/tree/main/src/Controller/Demo
 - https://github.com/rectorphp/getrector-com/tree/main/src/Demo

Not to speak on their behalf but as these are also PHP community projects I'm sure the maintainers may be able to help with guidance etc on making this possible

**Comment by Wirone** [comment]
In short: YES. It would be possible. There's only one problem: someone needs to build it 😅.

But seriously: the main culprit here is time, not the will or knowledge. It's also hard because only one person has full admin access in the org, so building complex processes involving several repositories or Github config may get long delays, which is a bit demotivating. Personally I've been thinking about Fixer playground for long, but just don't have time to work on it.
